### ISSUE TITLE
Accessibility: contrast too low

### ISSUE THREAD

**Issue by jrfnl** [gitextensions/gitextensions#9111]
## Current behaviour

Since upgrading to 3.5.0, it appears the default theme has been adjusted and the background for branch name labels has been changed.

This makes the text contrast too low, which makes reading these for anyone over 40 much harder.

## Expected behaviour

That the `default` theme is not changed on a whim, but put in a new theme. Or at the very least that changes take accessibility into account.

## Steps to reproduce

Upgrade from 3.4.3 to 3.5.0 with `default` theme.

## Screenshots

![image](https://user-images.githubusercontent.com/663378/115997680-590bbc80-a5e4-11eb-9526-a1935b749eeb.png)

## Did this work in previous version of GitExtensions

Yes.

## Environment

- Git Extensions 3.5.0.11713
- Build 0a5ef9ca6ce87754a510aa02c7a96d8b915714ac (Dirty)
- Git 2.31.1.windows.1
- Microsoft Windows NT 10.0.19042.0
- .NET Framework 4.8.4341.0
- DPI 144dpi (150% scaling)

**Comment by ivangrek** [comment]
Contrast or brightness?

**Comment by jrfnl** [comment]
I'd call it contrast, but then, I'm not a designer...

**Comment by mstv** [comment]
For me it is the contrast in the rows with the gray background and especially that it differs from row to row. The frame is distracting, too.

**Comment by jrfnl** [comment]
I think the removal/toning down of the background colour of the branch labels has a lot to do with it and I can't really see how I could customize this via a custom Theme (which I've started to create to undo the bug reported here: https://github.com/gitextensions/gitextensions/issues/9113).

**Comment by jrfnl** [comment]
Looks like this is related to/caused by #8900.

**Comment by RussKie** [comment]
@ivangrek the current colour scheme/contrast are likely failing the [minimum contrast test](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html).

For accessibility tests we use [Accessibility Insights](https://accessibilityinsights.io/docs/en/windows/overview/).
Use the "Color Contrast Analyzer" (3rd items in the menu) to find correct colours:
![image](https://user-images.githubusercontent.com/4403806/116024792-54054680-a692-11eb-84cb-d545e08111a8.png)

**Comment by ivangrek** [comment]
Thank you. I am research this situation.

**Comment by ivangrek** [comment]
### Base color
![image](https://user-images.githubusercontent.com/3169265/116308139-d5063000-a7af-11eb-940d-94cdd62ae92e.png)

### Color wheel
![image](https://user-images.githubusercontent.com/3169265/116308380-1565ae00-a7b0-11eb-8320-5dff6a614bf5.png)

### Color wheel (corrected)
![image](https://user-images.githubusercontent.com/3169265/116308712-85743400-a7b0-11eb-9884-9d734075676e.png)

### Red contrast
![image](https://user-images.githubusercontent.com/3169265/116307721-5c06d880-a7af-11eb-93d5-d5e62e7182be.png)

### Green contrast
![image](https://user-images.githubusercontent.com/3169265/116307775-6c1eb800-a7af-11eb-8b93-7ec6ed300e23.png)

### Result for light theme
![image](https://user-images.githubusercontent.com/3169265/116307676-4db8bc80-a7af-11eb-8235-dfd951f8f49a.png)

**Comment by ivangrek** [comment]
### Color wheel (corrected)
![image](https://user-images.githubusercontent.com/3169265/116310257-6d9daf80-a7b2-11eb-839d-32f62cb54b71.png)

### Red contrast
![image](https://user-images.githubusercontent.com/3169265/116309993-22839c80-a7b2-11eb-9d40-10cdf4bc4588.png)

### Green contrast
![image](https://user-images.githubusercontent.com/3169265/116309904-097aeb80-a7b2-11eb-9d12-ebf63e230eb3.png)

### Result for dark theme
![image](https://user-images.githubusercontent.com/3169265/116310346-8ad27e00-a7b2-11eb-92db-17efb357c114.png)

**Comment by mstv** [comment]
@ivangrek, also consider the alternating background, please. As a workaround, the branch labels could always be filled with the window color. This should work for `Highlight authored revisions`, too.

![image](https://user-images.githubusercontent.com/36601201/116311235-4ee7ea80-a7ab-11eb-8f21-53f5bb024397.png)

**Comment by ivangrek** [comment]
### Light
![image](https://user-images.githubusercontent.com/3169265/116312515-37156400-a7b5-11eb-992c-2b2a411e7569.png)

![image](https://user-images.githubusercontent.com/3169265/116312557-43012600-a7b5-11eb-9454-a83731e31592.png)

### Dark
![image](https://user-images.githubusercontent.com/3169265/116312656-662bd580-a7b5-11eb-92d7-a87331908af4.png)

![image](https://user-images.githubusercontent.com/3169265/116312697-7479f180-a7b5-11eb-8df8-5c51789e57df.png)

**Comment by ivangrek** [comment]
This is an experimental version with a rate of about 5. Most likely I will add more.

**Comment by jrfnl** [comment]
Another thing to take into account: Windows 10 "night light" setting which IIRC is on by default and tools like f.lux. Both dim the screen based on day/night and the contrast gets worse with the current colour choices when the screen is dimmed.

**Comment by gerhardol** [comment]
> This is an experimental version with a rate of about 5. Most likely I will add more.

The updates look fine to me. But I have not had a problem with the contrast for branches, occasionally with the select colors and alternate/highlight colors. Improving contrast for the labels may make the select colors worse.
This is hard...

**Comment by gerhardol** [comment]
@ivangrek Were you able to get better values?
I have been using the values you suggested, works OK for me.
This should be included in 3.5.1

```patch
diff --git a/GitUI/Themes/dark.css b/GitUI/Themes/dark.css
index 984f2..2ff59 100644
--- a/GitUI/Themes/dark.css
+++ b/GitUI/Themes/dark.css
@@ -38,9 +38,9 @@
 .HighlightAllOccurences      { color: #606020; }
 .Tag                         { color: #408080; }
 .Graph                       { color: #882d00; }
-.Branch                      { color: #00d76b; }
+.Branch                      { color: #3d9967; }
 .Branch.colorblind           { color: #0080ff; } /* hsl(210, 100%, 50%) */
-.RemoteBranch                { color: #ff8080; } /* hsl(0,   100%, 75%) */
+.RemoteBranch                { color: #de6459; } /* hsl(0,   100%, 75%) */
 .DiffSection                 { color: #4f4f4f; }
 .DiffRemoved                 { color: #4b0606; } /* hsl(0,   85%,  16%) */
 .DiffRemoved.colorblind      { color: #080646; } /* hsl(242, 84%,  15%) */
```

**Comment by gerhardol** [comment]
@jrfnl Are you OK with using the 3.4 colors in 3.5, is accessibility good enough?
Colors will not be switched back though, they will be similar to git-log.

**Comment by gerhardol** [comment]
Red and green colors are restored for the upcoming 3.5.1 (but colors are not switched back to 3.4, green is still local and red remote branch). Also the shading of the boxes is not done.

Someone with a good eye for design are welcome to tune the colors for contrast, both default and for the themes. 
Contrast should be tuned to selections etc too. Note that selection colors were tuned in #9277.

**Comment by jrfnl** [comment]
@gerhardol Appreciated and I look forward to seeing the change in action, though this means I will still need to use a custom theme, which turns out to be more problematic than expected (see #9166)
